lib/chef/data_collector.rb

```diff
@@ -376,6 +379,20 @@ def detect_unprocessed_resources
       def nested_resource?(new_resource)
         @current_resource_report && @current_resource_report.new_resource != new_resource
       end
+
+      def validate_data_collector_server_url!
+        raise Chef::Exceptions::ConfigurationError,
+          "data_collector.server_url is configured but empty. Please supply a valid URL." if data_collector_server_url.empty?
+
+        begin
+          uri = URI(data_collector_server_url)
+        rescue URI::InvalidURIError
+          raise Chef::Exceptions::ConfigurationError, "data_collector.server_url (#{data_collector_server_url}) is not a valid URI."
```

Comment:
Instead of `data_collector.server_url` in the error messages, how about `Chef::Config[:data_collector][:server_url]`? That would infer that it should be in the config file.